docs/performance-guide.md

```diff
@@ -25,150 +54,205 @@ guide intends to answer the following questions:
-| Inbound/Outbound Bandwidth (byte) | 2M    | 4M    | 10M   | 20M    | 40M    | 100M   | 200M    |
+| Inbound/Outbound (message/s) | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000 | 100,000 |
+| Inbound/Outbound Bandwidth (byte/s) | 2M    | 4M    | 10M   | 20M    | 40M    | 100M   | 200M    |
+
+In this use case, every client invokes the hub defined in the app server. The hub just calls the method defined in the original client side. This is the most light weighed hub for **echo**.
 
+```
+        public void Echo(IDictionary<string, object> data)
+        {
+            Clients.Client(Context.ConnectionId).SendAsync("RecordLatency", data);
+        }
+```
+
+Even for this simple hub, the traffic pressure on app server is also prominent as the **echo** inbound message increases. Therefore, it requires many app servers for big SKU tiers. The following table lists the app server count for every tier.
 
-App server count suggested
 
 |                  | Unit1 | Unit2 | Unit5 | Unit10 | Unit20 | Unit50 | Unit100 |
 |------------------|-------|-------|-------|--------|--------|--------|---------|
 | Connections      | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000 | 100,000 |
 | App server count | 2     | 2     | 2     | 3      | 3      | 10     | 20      |
 
+> **Note**
+>
+> The client connection number, message size, message sending rate, SKU tier and app server's CPU/Memory take impact on overall performance of **echo**.
 
 <a name="broadcast"></a>
 ### Broadcast
 
-For Broadcast, when web app receives the message, it broadcasts to all clients. More clients to broadcast, more message traffic.
+For **broadcast**, when web app receives the message, it broadcasts to all clients. More clients to broadcast, more message traffic to all clients. See the following diagram.
 
-The max inbound/outbound and bandwidth is summarized in the following table.
+<a name="broadcast_img"></a>
+![Broadcast](./images/broadcast.png)
+
+The characteristic of broadcast is there are a small number of clients broadcasting, which means the inbound message bandwidth is small, but the outbound bandwidth is huge. The outbound message increase as the client connection or broadcast rate increases.
+
+The maximum client connections, inbound/outbound message count and bandwidth are summarized in the following table.
 
 |                           | Unit1 | Unit2 | Unit5  | Unit10 | Unit20 | Unit50  | Unit100 |
 |---------------------------|-------|-------|--------|--------|--------|---------|---------|
 | Connections               | 1,000 | 2,000 | 5,000  | 10,000 | 20,000 | 50,000  | 100,000 |
-| Inbound (message/second)  | 4     | 2     | 2      | 2      | 2      | 2       | 2       |
-| Outbound (message/second) | 4,000 | 4,000 | 10,000 | 20,000 | 40,000 | 100,000 | 200,000 |
+| Inbound (message/s)  | 4     | 2     | 2      | 2      | 2      | 2       | 2       |
+| Outbound (message/s) | 4,000 | 4,000 | 10,000 | 20,000 | 40,000 | 100,000 | 200,000 |
```

Comment:
Any explanation why outbound of unit2 is same as unit1?

Reply:
Let me think about this.